net/golioth/Kconfig

```diff
@@ -39,6 +39,13 @@ config GOLIOTH_FW
 	  Enable Golioth firmware management, which allows to ask for newest
 	  desired firmware and issue firmware download process.
 
+config GOLIOTH_RPC_MAX_NUM_METHODS
```

Comment:
As this is the first commit, I thought that this is an option that is not used within the codebase. In fact, it is used starting from the second commit. It should be introduced together with `config GOLIOTH_RPC` and with dependency on `config GOLIOTH_RPC`, so that number of RPC methods would not be configurable with RPC being disabled.